pkg/beacon/relay/relay.go

```diff
@@ -148,10 +148,35 @@ func (n *Node) GenerateRelayEntry(
 				member.Signer,
 				startBlockHeight,
 			)
-			if err != nil {
+			if signatureErr != nil {
+				isEntryInProgress, err := relayChain.IsEntryInProgress()
+				if err != nil {
+					logger.Errorf(
+						"[member:%v] could not check entry status: [%v]; "+
+							"check was triggered because of threshold "+
+							"signature error: [%v]",
+						member.Signer.MemberID(),
+						err,
+						signatureErr,
+					)
+					return
+				}
+
+				// Check if we failed because someone else submitted in the
+				// meantime or because something wrong happened with
+				// our transaction.
+				if !isEntryInProgress {
```

Comment:
I think that this check should be inside of the submitter. This is submitter-specific logic (`submission.go`). One idea could be not to write to `errorChannel <- err` in this case.

Reply:
Yeah right, I changed.